File: class23_homework/A-Stanislav_Beregov-28/Plane.rb
$hash = Array.new(27) { |i| Array.new(2) { |i| 0 }}
def add_group gsize
	0.upto $hash.size-1 do |row|
		0.upto $hash[row].size-1 do |seats|
			if $hash[row][seats] <= 3-gsize
				$hash[row][seats] += gsize
				p $hash
				return true
			end
		end
	end
	return false
end

r = Random.new
loop do
	rand = r.rand(1..3)
	if add_group(rand)
		puts "successfully added a group of " + rand.to_s + " passengers!"
	else
		puts "Not enough space for a group of " + rand.to_s + " passengers!"
	end
	gets
end

File: class23_homework/A-Stanislav_Beregov-28/src/org/elsys/hw23/Plane.java
package org.elsys.hw23;

import java.io.IOException;
import java.util.Random;

public class Plane {

	static int[][] passengers = new int[27][2];
	
	static boolean addGroup(int gsize){
		for(int i = 0; i < 27; ++i){
			for(int a = 0; a < 2; ++a){
				if(passengers[i][a] <= 3-gsize){
					passengers[i][a] += gsize;
					return true;
				}
			}
		}
		return false;
	}
	
	public static void main(String[] args) throws IOException {
		Random random = new Random();
		while(true){
			 int rand = random.nextInt(3) + 1;
			 if(addGroup(rand))System.out.println("successfully added a group of " + rand + " passengers!");
			 else System.out.println("Not enough space for a group of " + rand + " passengers!");
			 System.in.read();
			 System.in.read();
		}
	}
}
